src/session.c

```diff
@@ -270,6 +271,8 @@ close_sessions_in_thread_func (GTask *task,
       session_close (session, FALSE);
     }
```

Comment:
Perhaps this function (`close_sessions_in_thread_func()`) should get a new name now that it does more things than closing sessions. Alternatively, add a dedicated step in `peer_died_cb()` after the other two.

Reply:
Went with the latter (the first version of this patch did exactly than before I moved it to `close_sessions_in_thread_func()`, it felt natural to move it back to `peer_died_cb()`)